lib/features/pirate_coins/domain/coins_domain.dart

```diff
@@ -1,14 +1,23 @@
 import "package:freezed_annotation/freezed_annotation.dart";
-import "package:flutter_riverpod/flutter_riverpod.dart";
-import "../data/coins_data.dart"; // Import the data layer file
+import "package:riverpod_annotation/riverpod_annotation.dart";
+
+import "../data/coins_data.dart";
+
 part "coins_domain.freezed.dart";
+part "coins_domain.g.dart";
 
+/// A number.
 @freezed
 abstract class NumberDomain with _$NumberDomain {
```

Comment:
Is there a better name for this? I'm thinking `Coins`. It needs to keep being usable with more data for it to hold added.

Reply:
Yeah... I agree... What should I rename where

Reply:
F2 it.

Reply:
Better name: `CoinsModel`.

Reply:
Also *note to self* I need to make `API` an `@freezed` class.